web_poet/fields.py

```diff
@@ -63,6 +73,28 @@ def __init__(self, method):
                 raise TypeError(
                     f"@field decorator must be used on methods, {method!r} is decorated instead"
                 )
+            if out:
+                orig_method = method
+                if inspect.iscoroutinefunction(method):
+
+                    @wraps(method)
+                    async def processed(*args, **kwargs):
+                        result = await orig_method(*args, **kwargs)
+                        for processor in out:
+                            result = processor(result)
+                        return result
+
+                else:
+
+                    @wraps(method)
+                    def processed(*args, **kwargs):
+                        result = orig_method(*args, **kwargs)
+                        for processor in out:
+                            result = processor(result)
+                        return result
+
+                method = processed
```

Comment:
A very small cleanup: define a function like this:

```py
def _process(out, result):  # out can be omitted if you defined it in a closure
    for processor in out:
        result = processor(result)
    return result
```

And then the code can be simplified a bit:
```suggestion
                if inspect.iscoroutinefunction(method):

                    @wraps(method)
                    async def processed(*args, **kwargs):
                        return _process(await orig_method(*args, **kwargs))
                        
                else:

                    @wraps(method)
                    def processed(*args, **kwargs):
                        return _process(orig_method(*args, **kwargs))
                        
                method = processed
```

Not sure if it worths it though :)

Reply:
Yeah, I thought about that too. Well, I think it simplifies the code, I'll implement it.

Reply:
You can also move the whole block to a separate function; it'd be similar to cached_method decorator. It looks fine as-is though.

Reply:
One more idea: use a single wraps() call (or update_wrapper), don't use a decorator

Reply:
e.g. `method = wraps(method)(processed)`

Reply:
done